pyaedt/modeler/circuits/PrimitivesNexxim.py

```diff
@@ -865,6 +865,55 @@ def create_voltage_pulse(self, compname=None, value_lists=[], location=[], angle
 
         return cmpid
 
+    @pyaedt_function_handler()
+    def create_voltage_pwl(
+        self, compname=None, time_list=[0], voltage_list=[0], location=[], angle=0, use_instance_id_netlist=False
+    ):
+        """Create a pwl voltage source.
+
+        Parameters
+        ----------
+        compname : str, optional
+            Name of the voltage pulse. The default is ``None``.
+        time_list : list, optional
+            List of time points for the pwl voltage source. The default is ``[0]``.
+        voltage_list : list, optional
+            List of voltages for the pwl voltage source. The default is ``[0]``.
+        location : list of float, optional
+            Position on the X axis and Y axis.
+        angle : float, optional
+            Angle rotation in degrees. The default is ``0``.
+        use_instance_id_netlist : bool, optional
+            Whether to use the instance ID in the net list.
+            The default is ``False``.
+
+        Returns
+        -------
+        :class:`pyaedt.modeler.cad.object3dcircuit.CircuitComponent`
+            Circuit Component Object.
+
+        References
+        ----------
+        >>> oEditor.CreateComponent
+        """
+        cmpid = self.create_component(
+            compname,
+            component_library="Independent Sources",
+            component_name="V_PWL",
+            location=location,
+            angle=angle,
+            use_instance_id_netlist=use_instance_id_netlist,
+        )
+
+        if len(time_list) != len(voltage_list):
+            raise ValueError("length of time not equal to length of voltage")
```

Comment:
```suggestion
            raise ValueError("Length of time is not equal to length of voltage.")
```
Messages should be complete sentences, with verbs and concluding punctuation.

Reply:
thanks!